azurerm/data_source_batch_account.go

```diff
@@ -0,0 +1,71 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmBatchAccount() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmBatchAccountRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMBatchAccountName,
+			},
+			"resource_group_name": resourceGroupNameDiffSuppressSchema(),
```

Comment:
whilst the Azure API's _technically_ state that the Resource Group name is case insensitive - unfortunately that's not the case in practice (since resources which rely on this may be) as such we need to make this case sensitive:

```suggestion
			"resource_group_name": resourceGroupNameSchema(),
```